controllers/horizontal_runner_autoscaler_webhook.go

```diff
@@ -716,31 +794,55 @@ func (autoscaler *HorizontalRunnerAutoscalerGitHubWebhook) SetupWithManager(mgr
 		switch hra.Spec.ScaleTargetRef.Kind {
 		case "", "RunnerDeployment":
 			var rd v1alpha1.RunnerDeployment
-
 			if err := autoscaler.Client.Get(context.Background(), types.NamespacedName{Namespace: hra.Namespace, Name: hra.Spec.ScaleTargetRef.Name}, &rd); err != nil {
+				autoscaler.Log.V(1).Info(fmt.Sprintf("RunnerDeployment not found with scale target ref name %s for hra %s", hra.Spec.ScaleTargetRef.Name, hra.Name))
 				return nil
 			}
 
-			keys := []string{rd.Spec.Template.Spec.Repository, rd.Spec.Template.Spec.Organization}
-
+			keys := []string{}
+			if rd.Spec.Template.Spec.Repository != "" {
+				keys = append(keys, rd.Spec.Template.Spec.Repository) // Repository runners
+			}
+			if rd.Spec.Template.Spec.Organization != "" {
+				if group := rd.Spec.Template.Spec.Group; group != "" {
+					keys = append(keys, rd.Spec.Template.Spec.Organization+"/group/"+rd.Spec.Template.Spec.Group) // Organization runner groups
```

Comment:
```suggestion
					keys = append(keys, organizationalRunnerGroupKey(organizatrd.Spec.Template.Spec.Organization, rd.Spec.Template.Spec.Group))
```

Would you mind extracting this logic to a `enterpriseKey`-like function so that it can be used from within `controllers/horizontal_runner_autoscaler_webhook.go` too and that makes future refactoring easier?

Reply:
good one - added in a7014dc9d80f2f874316e81ac886c2731d839e95